# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1485552181 18000
#      Fri Jan 27 16:23:01 2017 -0500
# Node ID 3b58b08d764a05ba94f4c43df43dd7c2380e355a
# Parent  2805cc365a86039d2f2a0777f202fe18aa740d65
delay cleanup for embedded interpreter until destructor is called

* interpreter.cc (interpreter::~interpreter):
Call cleanup if running in embedded mode.
(interpreter::execute): Call cleanup if not running in embedded mode.

diff --git a/libinterp/corefcn/interpreter.cc b/libinterp/corefcn/interpreter.cc
--- a/libinterp/corefcn/interpreter.cc
+++ b/libinterp/corefcn/interpreter.cc
@@ -538,16 +538,19 @@ namespace octave
 
     initialize_load_path (options.set_initial_path ());
 
     initialize_history (options.read_history_file ());
   }
 
   interpreter::~interpreter (void)
   {
+    if (m_embedded)
+      cleanup ();
+
     current_evaluator = 0;
 
     delete m_evaluator;
   }
 
   int interpreter::execute (void)
   {
     int exit_status = 0;
@@ -556,17 +559,18 @@ namespace octave
       {
         exit_status = execute_internal ();
       }
     catch (const octave::exit_exception& ex)
       {
         exit_status = ex.exit_status ();
       }
 
-    cleanup ();
+    if (! m_embedded)
+      cleanup ();
 
     return exit_status;
   }
 
   int interpreter::execute_internal (void)
   {
     display_startup_message ();
 
